### ISSUE TITLE
Article: A placeholder is not a label

### ISSUE THREAD

**Issue by davatron5000** [a11yproject/a11yproject.com#170]
Up for review! Comment over on the gist:

https://gist.github.com/davatron5000/5431990

**Comment by RachelRVasquez** [comment]
Wanted to drop a link here of the work in progress based off of what @davatron5000 has started: 

https://gist.github.com/RachelRVasquez/9aed0ddda6eaf675fd15

Will finish this when I have a chance, but open for suggestions so far. :-)

**Comment by gtarnoff** [comment]
Looking good. It might just be me, but the last paragraph seems like it should be the second. You state that placeholders don't work for screen readers, then it's ok to keep using, then how to do it properly. Does that make sense?

**Comment by RachelRVasquez** [comment]
@gtarnoff That's true! It does make sense to mention placeholders can still be used so long as they're used with labels rather than a replacement - before going into styling explanations. Will make the change when I get a chance. Thanks! :-)

**Comment by pstonier** [comment]
Looks great. I agree with all the comments above.

**Comment by RachelRVasquez** [comment]
@pstonier @gtarnoff @davatron5000 

Okay looks like I got around to wrapping it up. Open for more feedback. :-)

**Comment by gtarnoff** [comment]
Only thing I have to comment on is capitalizing CSS. The actual term is an acronym and should be in all caps.

**Comment by thierryk** [comment]
@RachelRVasquez: a few thoughts:

   * Wording: 
      * I'd change the title from "How to properly use a placeholder in a form" to "Using @placeholder on input" and may be link to [the spec](http://www.w3.org/WAI/GL/wiki/Using_@placeholder_on_input) somehow.
      * I'd change the name of the section "Label as a wrapper" for "Implicit labelling"
      * I'd change the name of the section "Label using the "for" attribute" for "explicit labelling"
      * I'd use the term _form control_ instead of form field 
      * As suggested by @gtarnoff CSS is an abbreviation and should be written in uppercase
   * Technique:
      * I'd show the use of `@for` in the implicit labelling example and indicate that even if it is valid to omit `@for` in this case, it is **best practice** to include it (see [this thread](https://twitter.com/fstorr/status/674647841386586114)). 
   * Examples:
      * I would suggest that because of backward compatibility, it should be safe to only use the former syntax for Firefox: `:-moz-placeholder { color: #626262; }`
      * I'd drop the `/` in void elements (using `<input>` instead of `<input/>`

**Comment by RachelRVasquez** [comment]
@gtarnoff @thierryk 

Thanks for the feedback guys! I'll get to working those fixes in some time today hopefully. (fingers crossed) :-)

**Comment by RachelRVasquez** [comment]
@thierryk 

Do you have any resources that explain why you call the code samples, Implicit/Explicit labeling? I've never heard the terms before so I imagine there might be others that haven't either. So having something to reference may help both me and others understand better. :-)

I also took a look at the W3 article and decided to use their term "input element" instead of form field. I also adjusted the title to the way they explain it as @placeholder was confusing for me at first until I read the article.

**Comment by RachelRVasquez** [comment]
I've made further changes for review based on the suggestions so far.

https://gist.github.com/RachelRVasquez/9aed0ddda6eaf675fd15

**Comment by thierryk** [comment]
@RachelRVasquez 

> Do you have any resources that explain why you call the code samples, Implicit/Explicit labeling?

I believe those terms (implicit/explicit labelling) are widely used in a11y the community.
They are not technical terms. _Implicit_ means something is *implied*&mdash;while _explicit_ means something is *clearly expressed*. 

In our case:
   * the nesting *implies* that the `label` is associated with the form control it contains
   * the use of `@for` associates *explicitly* the 2 elements

As a side note, I mentioned earlier a link to a thread that explains the reasons why the latter is considered best practice. I think you may want to reference that or at least explain why it is better to rely on `@for`: it is related to UA support so some day, this will _not be considered best practice anymore_.

**Comment by RachelRVasquez** [comment]
@thierryk 

Cool - thanks for explaining that to me. I read the article associated to that thread you linked to as well and it helped me understand the terms better. I've tried to summarize what the terms mean in my gist a bit more without going into as much detail as the article. So I linked the article for a more in-depth explanation if people need to go that route.

I also re-named the former "implicit" example as a combination of implicit/explicit since it combines the two techniques. :-)

**Comment by RachelRVasquez** [comment]
@davatron5000 I think this one is ready to "article-ize" as well. Unless you have some other feedback? Since you were the original developer on the topic - definitely could use your input on whether this went in the direction you intended. Thanks! :-)

**Comment by davatron5000** [comment]
:+1: This is great too! My only nit pick would be to change:

```
You can make your forms accessible by following these steps:
```

to something like:

```
Make your forms accessible by using one of the following methods (in order of preference):
```

`label` + `aria-labelledby` + hidden `label` would be somewhat redundant. Also the "include a proper `label`" should probably be "include a proper visible `label`" since that would be most accessible to everyone.

**Comment by thierryk** [comment]
> Also the "include a proper label" should probably be "include a proper visible label" since that would be most accessible to everyone.

It depends on what is a "label" in this context. If we're talking `<label>` then no, I don't think it is necessary to make it visible as long as there is some other visual cue (i.e. the use of a `@placeholder`&mdash;with proper contrast).

**Comment by RachelRVasquez** [comment]
@davatron5000 Ah ok I understand. So we're treating the checklist as people don't have to do them all, just use one of the methods listed. So if they're using a visible label, `aria-label` and `aria-labelledby` would be repetitive. If they're visually hiding a label (accessibly), then having a visual cue like placeholder with proper contrast will work fine.

I've made the edits to the article.

**Comment by svinkle** [comment]
Closed via #386.
